### PR TITLE
Fix quickstart config and add regression tests

### PR DESCRIPTION
**Pull request by ekneg54**
Had to revert this change and create a new PR because it catched a bug in the main branch which possibly were introduced with #86 and will be fixed with this PR.

### CODE REVIEW

**Comment by ekneg54** [comment]
rebased main and pushed to fix regression of pipeline jobs

File: tests/unit/test_run_logprep.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    @mock.patch("logprep.run_logprep._run_logprep")
+    def test_main_calls_run_logprep_with_quickstart_config(self, mock_run_logprep):
+        """ensures the quickstart config is valid"""
+        sys.argv = ["logprep", "--disable-logging", "quickstart/exampledata/config/pipeline.yml"]
+        run_logprep.main()
+        mock_run_logprep.assert_called()
+
+    @mock.patch("logprep.util.schema_and_rule_checker.SchemaAndRuleChecker.validate_rules")
+    def test_main_calls_validates_rules(self, mock_validate_rules):
+        """ensures the quickstart config is valid"""
```
Comment (dtrai2) [inline tests/unit/test_run_logprep.py:20]:
```suggestion
        """ensures rule validation is called"""
```

File: tests/unit/test_run_logprep.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            "--disable-logging",
+            "--validate-rules",
+            "quickstart/exampledata/config/pipeline.yml",
+        ]
+        with pytest.raises(SystemExit):
+            run_logprep.main()
+        mock_validate_rules.assert_called()
+
+    def test_quickstart_rules_are_valid(self):
+        """ensures the quickstart config is valid"""
```
Comment (dtrai2) [inline tests/unit/test_run_logprep.py:32]:
```suggestion
        """ensures the quickstart rules are valid"""
```

**Review by dtrai2 — APPROVED** [review]

**Review by urdit52 — APPROVED** [review]
looks good to me.

**Pull request merged**